sql/partition_info.cc

```diff
@@ -1211,6 +1211,7 @@ bool partition_info::vers_setup_2(THD * thd, bool is_create_table_ind)
         }
         else if (vers_scan_min_max(thd, el))
         {
+          table->s->stat_trx= NULL; // may be a leak on endless table open
```

Comment:
No, it fixes open error situation (impossible to test). `stat_trx` is indicator for `dont_stat`:
```c++
    if (!table->s->stat_trx)
    {
      DBUG_ASSERT(partitions.elements > 1);
      table->s->stat_trx= static_cast<Vers_field_stats**>(
        alloc_root(&table->s->mem_root, sizeof(void *) * partitions.elements * num_columns));
      dont_stat= false;
    }
```
Once set, `stat_trx` prevents idempotency of `vers_scan_min_max` error. Thus, we need to clear it. Otherwise we get error only once and afterwards open will not fail leading to wrong table usage.